src/librbd/internal.cc

```diff
@@ -1557,6 +1557,13 @@ int invoke_async_request(ImageCtx *ictx, const std::string& request_type,
       r = -ENOSYS;
       goto err_close_parent;
     }
+    
+    if (r < 0) {
+      // we lost the race with snap removal?
+      lderr(cct) << "parent SnapInfo is gone" << dendl;
```

Comment:
Users won't know what ```SnapInfo``` is -- suggest something like "unable to locate parent snapshot".  Should also probably return -ENOENT instead of -EINVAL.